### ISSUE TITLE
Request for VLind-Bench reproduction details for CausalMM on LLaVA-1.5

### ISSUE THREAD

**Issue by piiiiiila** [The-Martyr/CausalMM#10]
Thanks for releasing the CausalMM code.
I am trying to reproduce the VLind-Bench results reported in the paper, especially the LLaVA-1.5 results in Figure 3. However, I could not find the exact VLind-Bench reproduction script or hyperparameter settings in the current repository.

My setup:

- Model: LLaVA-v1.5-7B
- CausalMM repo: current released code
- VLind-Bench dataset: `klee972/VLind-Bench`
- Environment:
  - Python 3.9
  - PyTorch 2.4.1 + CUDA 12.1
  - transformers 4.31.0
- CausalMM code path:
  - `llava-1.5/experiments/llava/model/multimodal_encoder/cf_encoder.py`
  - `llava-1.5/experiments/llava/model/language_model/llava_llama_cf.py`
  - `llava-1.5/causalmm_cf/causalmm_sm.py`

I implemented a VLind-Bench evaluator following the official VLind-Bench evaluation pipeline. For each instance, I run the a/b/c/d True-False questions and save predictions in the same format expected by `score_pipeline.py`.

However, using the released CausalMM LLaVA code with the POPE-style default parameters:

```bash
gamma=1.0
epsilon=0.6
seed=55
temperature=1.0
top_p=1.0
```

the model tends to answer almost all VLind queries as `True`. On the full VLind-Bench run, I got:

```text
causalmm scores:
3.3 & 1.0 & 40.0 & 0.0 & 18.5 & 3.3
```

I also checked a 20-sample subset. The answer distribution was:

```text
a1_image: {'True': 20}
a2_image: {'False': 2, 'True': 18}
b1: {'True': 20}
b2: {'True': 20}
c1_image: {'True': 20}
c2_image: {'True': 20}
d1: {'True': 175}
d2: {'True': 175}
```

So the low score seems to come from a strong True bias in the generated True/False answers, especially for the negative queries `b2`, `c2`, and `d2`.

Could you please provide the missing VLind-Bench reproduction details?

Specifically:

1. What exact script was used to evaluate CausalMM on VLind-Bench?
2. What prompt templates were used for LLaVA-1.5 on VLind-Bench?
3. Did you use free-form generation, constrained True/False decoding, or logits-based forced-choice scoring?
4. What exact decoding parameters were used?
   - `gamma`
   - `epsilon`
   - `temperature`
   - `top_p`
   - `top_k`
   - `max_new_tokens`
   - random seed
5. In Figure 3, what code paths correspond to the Vision, Language, and Multimodal CausalMM variants?
6. For the LLaVA-1.5 VLind-Bench result, did you use only visual counterfactual attention, only language counterfactual attention, or both?
7. Is the visual counterfactual attention method the default `shuffle` in `cf_encoder.py`, or another method such as `random`, `uniform`, or `reversed`?
8. Was any post-processing applied to model outputs, such as mapping `Yes/No` to `True/False`?
9. Could you release the VLind-Bench evaluation script and/or the prediction JSON files used for Figure 3?

These details would be very helpful for reproducing the VLind-Bench results faithfully.

Thanks!

**Comment by The-Martyr** [comment]
Unfortunately, my specific data was bundled into a massive archive and stored in the cloud during a cluster migration. I currently have nowhere to extract it.

Although some phrasing in your ai-style question was a bit confusing to me, I will do my best to help. Regarding the code, we did not use any additional prompts. You should try greedy decoding. As for the LLaVA-1.5 results on VLind, our paper covers all three settings, whereas the current code only supports the "visual-only" setting. I recall that shuffle mode yielded the best results, but not for sure though.

Assuming your experimental setup is correct, the key factors should be gemma and epsilon. The decoding process in CausalMM relies on a rather delicate balance. since it requires no training, I suggest performing a simple hyperparameter search within the 1.0 range through vibe coding. You might also well achieve metrics superior to those reported in our paper.
